File: examples/go/user_service.go
package service

import (
	"errors"
	"regexp"

	"testmind/models"
	"testmind/repositories"
)

var emailRegex = regexp.MustCompile(`^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$`)

type UserService struct {
	userRepository repositories.UserRepository
}

func NewUserService(userRepo repositories.UserRepository) *UserService {
	return &UserService{
		userRepository: userRepo,
	}
}

func (s *UserService) FindUserByID(id int64) (*models.User, error) {
	if id <= 0 {
		return nil, errors.New("用户ID必须为正数")
	}

	return s.userRepository.FindByID(id)
}

func (s *UserService) FindAllUsers() ([]*models.User, error) {
	return s.userRepository.FindAll()
}

func (s *UserService) CreateUser(user *models.User) (*models.User, error) {
	if user == nil {
		return nil, errors.New("用户信息不能为空")
	}

	if user.Email == "" {
		return nil, errors.New("用户邮箱不能为空")
	}

	if err := s.validateEmailFormat(user.Email); err != nil {
		return nil, err
	}

	return s.userRepository.Save(user)
}

func (s *UserService) UpdateUser(id int64, user *models.User) (*models.User, error) {
	if id <= 0 {
		return nil, errors.New("用户ID必须为正数")
	}

	existingUser, err := s.userRepository.FindByID(id)
	if err != nil {
		return nil, err
	}
	if existingUser == nil {
		return nil, errors.New("用户不存在")
	}

	return s.userRepository.Save(user)
}

func (s *UserService) DeleteUser(id int64) error {
	if id <= 0 {
		return errors.New("用户ID必须为正数")
	}

	return s.userRepository.DeleteByID(id)
}

func (s *UserService) validateEmailFormat(email string) error {
	if !emailRegex.MatchString(email) {
		return errors.New("邮箱格式不正确")
	}
	return nil
}

File: examples/java/UserService.java
package com.testmind.service;

import com.testmind.model.User;
import com.testmind.repository.UserRepository;
import java.util.List;
import java.util.Optional;

/**
 * 用户服务类
 * 提供用户管理的核心功能
 */
public class UserService {

    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    /**
     * 根据ID查找用户
     */
    public Optional<User> findUserById(Long id) {
        if (id == null || id <= 0) {
            throw new IllegalArgumentException("用户ID必须为正数");
        }
        return userRepository.findById(id);
    }

    /**
     * 获取所有用户
     */
    public List<User> findAllUsers() {
        return userRepository.findAll();
    }

    /**
     * 创建新用户
     */
    public User createUser(User user) {
        if (user == null) {
            throw new IllegalArgumentException("用户信息不能为空");
        }
        if (user.getEmail() == null || user.getEmail().trim().isEmpty()) {
            throw new IllegalArgumentException("用户邮箱不能为空");
        }
        validateEmailFormat(user.getEmail());
        return userRepository.save(user);
    }

    /**
     * 更新用户信息
     */
    public User updateUser(Long id, User user) {
        if (id == null || id <= 0) {
            throw new IllegalArgumentException("用户ID必须为正数");
        }
        Optional<User> existingUser = userRepository.findById(id);
        if (existingUser.isEmpty()) {
            throw new RuntimeException("用户不存在");
        }
        return userRepository.save(user);
    }

    /**
     * 删除用户
     */
    public void deleteUser(Long id) {
        if (id == null || id <= 0) {
            throw new IllegalArgumentException("用户ID必须为正数");
        }
        userRepository.deleteById(id);
    }

    /**
     * 验证邮箱格式
     */
    private void validateEmailFormat(String email) {
        String emailRegex = "^[A-Za-z0-9+_.-]+@(.+)$";
        if (!email.matches(emailRegex)) {
            throw new IllegalArgumentException("邮箱格式不正确");
        }
    }
}
